torchgeo/models/changestar.py

```diff
@@ -0,0 +1,230 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""ChangeStar implementations."""
+
+from typing import Any, Dict, List
+
+import torch
+from einops import rearrange
+from torch import Tensor
+from torch.nn.modules import (
+    BatchNorm2d,
+    Conv2d,
+    Identity,
+    Module,
+    ReLU,
+    Sequential,
+    UpsamplingBilinear2d,
+)
```

Comment:
I would use `import torch.nn as nn` and `nn.BatchNorm2d`, etc. for these too just like with the tests.

Reply:
such as nn.modules.BatchNorm2d?

Reply:
such as `nn.modules.BatchNorm2d`?

Reply:
Seems like directly using nn.BatchNorm2d will raise mypy issues.

Reply:
I think that's a bug in PyTorch's type hints (or maybe in mypy) so you can either use `nn.modules.BatchNorm2d` or use `nn.BatchNorm2d` and ignore the mypy error with a comment.

Reply:
torchvision has a [mypy configuration file](https://github.com/pytorch/pytorch/blob/master/mypy.ini).
Maybe, adding this file can solve these mypy issues fundamentally.

Reply:
We also have some configuration in https://github.com/microsoft/torchgeo/blob/main/pyproject.toml#L44

I like the idea of disabling some of these warnings in configuration. I probably won't get to this in the next couple of weeks, so use whatever you want for now and I can change it later.

Reply:
I had similar problems with type hints when adding other change detection models. There are workarounds that you can reference here https://github.com/microsoft/torchgeo/blob/main/torchgeo/models/fccd.py.

Reply:
Thanks, that's very helpful.